app/models/tenant.rb

```diff
@@ -155,6 +155,18 @@ def login_logo?
     !!login_logo_file_name
   end
 
+  def visible_domains
+    MiqAeDomain.where(:tenant_id => ancestor_ids.append(id)).order('priority DESC')
```

Comment:
We want to include ancestry while sorting here:

Should be something like this:
```ruby
MiqAeDomain.where(:tenant_id => ancestor_ids.append(id)).join('tenants on miq_ae_domains.tenant_id = tenants.id').order('tenants.ancestry ASC, priority DESC')
```